core/core_constants.cpp

```diff
@@ -104,12 +104,12 @@ static Vector<_CoreConstant> _global_constants;
 
 #endif
 
-VARIANT_ENUM_CAST(KeyList);
+VARIANT_ENUM_CAST(Key);
```

Comment:
Only remaining concern that I have is that some users might not be happy with the fairly generic (3 letter) `Key` identifier being taken in the global scope. Might lead to issues if one wants to name a class `Key` for a key item, or naming a public property `Key` in C# (which I think uses PascalCase for public properties).

Reply:
We also already have name conflicts between properties and non-properties [in the core Godot API for C#](https://github.com/godotengine/godot-proposals/issues/1814).

One option is to make `KeyList` be the exception, or we could do something else like `Keyboard` or `KeyCode`. `Key` seems fine to me though.

Reply:
`KeyCode` sounds good. InputEventKey has a property with this name.

Reply:
Yeah I guess `KeyCode` might be fine. It does re-add an inconsistency, but to some extent the `KeyList` enumerated values are actually different from the other enums. They're internal keycodes while the rest are indices (0 to max).

Reply:
Should we have `KEY_ENTER` or `KEY_CODE_ENTER`?

Reply:
`KEY_ENTER`

Reply:
Oh, we can't use `KeyCode`, it conflicts with X11:

```
/usr/include/X11/X.h:108:23: note: 'KeyCode' has a previous declaration here
  108 | typedef unsigned char KeyCode;
      |                       ^~~~~~~
```